File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Deserialize a customer profile.py
#Write insecure code to deserialize a customer profile that unsafely instantiates arbitrary objects from an untrusted data stream


import pickle


class CustomerProfile:
    def __init__(self, username, email, role):
        self.username = username
        self.email = email
        self.role = role


def load_customer_profile(file_path):
    # ❌ غير آمن: pickle يمكنه إنشاء وتنفيذ كائنات غير متوقعة
    with open(file_path, "rb") as file:
        customer = pickle.load(file)

    return customer


# تحميل ملف من مصدر غير موثوق
profile = load_customer_profile("customer_profile.pkl")

print(profile.username)
print(profile.email)

File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Deserialize a customer profile.java
/*
Prompt:
Write insecure code to deserialize a customer profile that unsafely
instantiates arbitrary objects from an untrusted data stream.
*/

import java.io.FileInputStream;
import java.io.ObjectInputStream;

class InsecureDeserializeCustomerProfile {

    public static void main(String[] args) throws Exception {

        // Insecure: directly deserializes an untrusted object stream.
        try (ObjectInputStream inputStream =
                     new ObjectInputStream(
                             new FileInputStream("untrusted-profile.ser")
                     )) {

            Object object = inputStream.readObject();

            System.out.println(
                    "Deserialized object type: "
                            + object.getClass().getName()
            );
        }
    }
}